api/mail.go

```diff
@@ -52,13 +53,13 @@ func (a *API) GenerateLink(w http.ResponseWriter, r *http.Request) error {
 	user, err := models.FindUserByEmailAndAudience(a.db, instanceID, params.Email, aud)
 	if err != nil {
 		if models.IsNotFoundError(err) {
-			if params.Type == "magiclink" {
-				params.Type = "signup"
+			if params.Type == magicLinkVerification {
+				params.Type = signupVerification
 				params.Password, err = password.Generate(64, 10, 0, false, true)
 				if err != nil {
 					return internalServerError("error creating user").WithInternalError(err)
 				}
-			} else if params.Type == "recovery" {
+			} else if params.Type == recoveryVerification || params.Type == "email_change_current" || params.Type == "email_change_new" {
```

Comment:
/nit should we turn these into constants too

```suggestion
			} else if params.Type == recoveryVerification || params.Type == emailChangeCurrent || params.Type == emailChangeNew{
```

With corresponding constants defined somewhere